stacks/config.js

```diff
@@ -150,14 +150,15 @@ export function setupSentry (app, stack) {
 
 /**
  * @param {import('sst/constructs').Stack} stack
+ * @param  {{domainName: string, hostedZone: string} | undefined} customDomain
  */
-export function getServiceURL (stack) {
+export function getServiceURL (stack, customDomain) {
   // in production we use the top level subdomain
   if (stack.stage === 'prod') {
     return 'https://up.web3.storage'
-  // in staging and PR environments we use a sub-subdomain
-  } else if (stack.stage === 'staging' || stack.stage.startsWith('pr')) {
-    return `https://${stack.stage}.up.web3.storage`
+  // Derive from custom domain if there is one, which is used in staging, PR envs and dev envs
+  } else if (customDomain) {
+    return customDomain.domainName
```

Comment:
this way we also include dev setups, while supporting staging and PRs. Only when a custom domain is not set, we need to go via env var